runtime/src/iree/hal/drivers/hip/event_pool.c

```diff
@@ -251,15 +251,33 @@ iree_status_t iree_hal_hip_event_pool_acquire(
   // Allocate the rest of the events.
   if (remaining_count > 0) {
     IREE_TRACE_ZONE_BEGIN_NAMED(z1, "event-pool-unpooled-acquire");
-    iree_status_t status = iree_ok_status();
-    for (iree_host_size_t i = 0; i < remaining_count; ++i) {
-      status = iree_hal_hip_event_create(
-          event_pool->symbols, event_pool, event_pool->hip_context,
-          event_pool->host_allocator, &out_events[from_pool_count + i]);
+
+    iree_status_t status = IREE_HIP_CALL_TO_STATUS(
+        event_pool->symbols, hipCtxPushCurrent(event_pool->device_context),
+        "hipCtxPushCurrent");
+    if (iree_status_is_ok(status)) {
+      for (iree_host_size_t i = 0; i < remaining_count; ++i) {
+        status = iree_hal_hip_event_create(event_pool->symbols, event_pool,
+                                           event_pool->host_allocator,
+                                           &out_events[from_pool_count + i]);
+        if (!iree_status_is_ok(status)) {
+          // Must release all events we've acquired so far.
+          iree_hal_hip_event_pool_release_event(event_pool, from_pool_count + i,
+                                                out_events);
+          IREE_TRACE_ZONE_END(z1);
+          IREE_TRACE_ZONE_END(z0);
```

Comment:
you don't need to end z0 here as you break, but you do need to end it below because you return - you should probably return here too